Sources/Common/Core/DataArray/index.js

```diff
@@ -262,9 +262,14 @@ function vtkDataArray(publicAPI, model) {
   };
 
   publicAPI.setData = (typedArray, numberOfComponents) => {
-    model.values = typedArray;
-    model.size = typedArray.length;
-    model.dataType = getDataType(typedArray);
+    if (typedArray) {
+      model.values = typedArray;
+      model.size = typedArray.length;
+      model.empty = model.size === 0;
```

Comment:
Probably not the right test to do.
If initialValues.values is null, then you define an empty typedArray of size = model.size.
publicAPI.setData is called afterwards, so model.empty will be set to false whereas it should be true.

Reply:
I updated the function to correspond to the behavior it had before (throw an error if typedArray is either null or undefined).